### PR TITLE
Fixes #18223: Add a page for technical stack doc

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/18223

### CODE REVIEW

File: src/reference/modules/ROOT/pages/technical_stack.adoc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+which has static typing and focuses on reliability.
+
+=== Relay
+
+The second server service is the relay, which handles communication with the managed machines.
+It can be installed without the webapp part (and then be used as a kind of smart proxy).
+
+image::tech_stack/rust.svg[Rust,height=200]
+
+It is written in *https://www.rust-lang.org/[Rust]* (using the https://tokio.rs/[tokio] asynchronous runtime) to allow great performance even on very small hardware while providing to provide security,
```
Comment (fanf) [inline src/reference/modules/ROOT/pages/technical_stack.adoc:60]: providing to provide security

File: src/reference/modules/ROOT/pages/technical_stack.adoc [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+** Avoid https://en.wikipedia.org/wiki/Not_invented_here[NIH]: use the existing tools that does the job well, especially for things out of our core
+business (web server, data storage, etc.). Use standard system components whenever possible, especially for the agent.
+
+** For components we integrate (programs or libraries), this means choosing a mature and reliable
+option, even more for the agent.
+
+* Agent and relay components need to be able to run on very light systems with low network requirements,
+allowing embedded use cases
+
+* Stay as simple as possible, particularly on the operating side. Users should not have to spend
```
Comment (fanf) [inline src/reference/modules/ROOT/pages/technical_stack.adoc:27]: I think we need to add here somewhere: "this also means limiting the number of technologies in one given component"

File: src/reference/modules/ROOT/pages/technical_stack.adoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+and using official python APIs for some of the integrated tools.
+
+=== Databases
+
+There are three types of data managed by the server:
+
+* All the configuration data (configuration policies, groups, etc.) are stored in a *git* repository.
+It provides traceability over the configuration history and interoperability with external SCM.
+* The data generated from the configuration (reports, etc.) is stored in *PostgreSQL*, which
+is mature and reliable, and also provides
```
Comment (fanf) [inline src/reference/modules/ROOT/pages/technical_stack.adoc:81]: missing end of sentence

**Review by fanf — CHANGES_REQUESTED** [review]

**Comment by amousset** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**